From 9b83afdc667dabf2bcc8a72024cc4ae0a414f93a Mon Sep 17 00:00:00 2001
From: "A. Maitland Bottoms" <bottoms@debian.org>
Date: Sat, 10 Feb 2018 19:50:21 -0500
Subject: [PATCH 03/27] bsd hurd alsa

Debian's kFreeBSD and Hurd kernels have alsa sound, but do
not have the Linux kernel's ESTRPIPE.
---
 gr-audio/lib/CMakeLists.txt      | 4 ++--
 gr-audio/lib/alsa/alsa_sink.cc   | 4 ++--
 gr-audio/lib/alsa/alsa_source.cc | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/gr-audio/lib/CMakeLists.txt b/gr-audio/lib/CMakeLists.txt
index 944b1599b1..72795ed744 100644
--- a/gr-audio/lib/CMakeLists.txt
+++ b/gr-audio/lib/CMakeLists.txt
@@ -46,7 +46,7 @@ endif(ENABLE_GR_CTRLPORT)
 ########################################################################
 find_package(ALSA)
 
-if(ALSA_FOUND)
+if((ALSA_FOUND) AND NOT (CMAKE_SYSTEM_NAME STREQUAL "kFreeBSD") AND NOT (CMAKE_SYSTEM_NAME STREQUAL GNU))
 
     add_definitions(-DALSA_FOUND)
     GR_APPEND_SUBCOMPONENT("alsa")
@@ -61,7 +61,7 @@ if(ALSA_FOUND)
     )
     list(APPEND gr_audio_confs ${CMAKE_CURRENT_SOURCE_DIR}/alsa/gr-audio-alsa.conf)
 
-endif(ALSA_FOUND)
+endif((ALSA_FOUND) AND NOT (CMAKE_SYSTEM_NAME STREQUAL "kFreeBSD") AND NOT (CMAKE_SYSTEM_NAME STREQUAL GNU))
 
 ########################################################################
 ## OSS Support
diff --git a/gr-audio/lib/alsa/alsa_sink.cc b/gr-audio/lib/alsa/alsa_sink.cc
index f2e70ca2d2..9695ad2c71 100644
--- a/gr-audio/lib/alsa/alsa_sink.cc
+++ b/gr-audio/lib/alsa/alsa_sink.cc
@@ -507,7 +507,7 @@ namespace gr {
           }
           continue;  // try again
         }
-
+#ifdef ESTRPIPE
         else if(r == -ESTRPIPE) {  // h/w is suspended (whatever that means)
 				   // This is apparently related to power management
           d_nsuspends++;
@@ -517,7 +517,7 @@ namespace gr {
           }
           continue;  // try again
         }
-
+#endif
         else if (r < 0) {
           output_error_msg("snd_pcm_writei failed", r);
           return false;
diff --git a/gr-audio/lib/alsa/alsa_source.cc b/gr-audio/lib/alsa/alsa_source.cc
index 2cfc33dd44..20b5417c73 100644
--- a/gr-audio/lib/alsa/alsa_source.cc
+++ b/gr-audio/lib/alsa/alsa_source.cc
@@ -472,7 +472,7 @@ namespace gr {
           }
           continue;  // try again
         }
-
+#ifdef ESTRPIPE
         else if(r == -ESTRPIPE) {   // h/w is suspended (whatever that means)
                                     // This is apparently related to power management
           d_nsuspends++;
@@ -482,7 +482,7 @@ namespace gr {
           }
           continue;   // try again
         }
-
+#endif
         else if(r < 0) {
           output_error_msg("snd_pcm_readi failed", r);
           return false;
-- 
2.11.0

